dotnet/src/SemanticKernel.Core/Data/TextSearch/VectorStoreTextSearch.cs

```diff
@@ -152,6 +153,46 @@ public async Task<KernelSearchResults<object>> GetSearchResultsAsync(string quer
     private readonly IVectorizableTextSearch<TRecord>? _vectorizableTextSearch;
     private readonly ITextSearchStringMapper _stringMapper;
     private readonly ITextSearchResultMapper _resultMapper;
+    private readonly Lazy<TextSearchResultPropertyReader> _propertyReader;
+
+    /// <summary>
+    /// Result mapper which converts a TRecord to a <see cref="TextSearchResult"/>.
+    /// </summary>
+    private TextSearchResultMapper CreateTextSearchResultMapper()
+    {
+        TextSearchResult MapFromResultToTextSearchResult(object result)
+        {
+            if (typeof(TRecord) != result.GetType())
+            {
+                throw new ArgumentException($"Expected result of type {typeof(TRecord).FullName} but got {result.GetType().FullName}.");
+            }
+
+            return new TextSearchResult(
+                name: this._propertyReader.Value.GetName(result),
+                value: this._propertyReader.Value.GetValue(result),
+                link: this._propertyReader.Value.GetLink(result));
+        };
+
+        return new TextSearchResultMapper(MapFromResultToTextSearchResult);
+    }
+
+    /// <summary>
+    /// /// Result mapper which converts a TRecord to a <see cref="string"/>.
+    /// </summary>
+    private TextSearchStringMapper CreateTextSearchStringMapper()
+    {
+        string MapFromResultToString(object result)
+        {
+            if (typeof(TRecord) != result.GetType())
+            {
+                throw new ArgumentException($"Expected result of type {typeof(TRecord).FullName} but got {result.GetType().FullName}.");
+            }
+
+            var value = this._propertyReader.Value.GetValue(result);
+            return (string?)value ?? throw new InvalidOperationException("Value property cannot be null.");
+        };
```

Comment:
Why we have a inner method that pretty much is the whole method body, I think for those scenarios make sense just merge the logic in on method if it's not used anywhere else.